test/e2e/e2e_test.go

```diff
@@ -609,38 +587,9 @@ var _ = Describe("e2e multiple external gateway update validation", func() {
 		framework.ExpectNoError(err)
 
 		// Wait for pod exgw setup to be almost ready
-		wait.PollImmediate(1*time.Second, 30*time.Second, func() (bool, error) {
-			jsonFlag := "jsonpath='{.metadata.annotations.k8s\\.ovn\\.org/hybrid-overlay-external-gw}'"
-			kubectlOut, err := framework.RunKubectl("get", "pod", ovnNsFlag, srcPingPodName, jsonFlag)
-			if err != nil {
-				return false, nil
-			}
-			if !strings.Contains(kubectlOut, extGwAlt1) {
```

Comment:
This does not have the same "variable was never initialized" problem that the other example you pointed out does, so it looks like this is actually removing a check.

Reply:
I explained myself wrongly, sorry, the "empty variable" was only one thing, but the important want is that this checks are inside a loop that can fail and is never asserted. 
```
// Wait for pod exgw setup to be almost ready			// setup the first external gateway
		wait.PollImmediate(1*time.Second, 30*time.Second, func() (bool, error) {
``` 
because if the checks fail, wait.PollImmediate will return an error, and we are never asserting it. Will recover the checks because I saw in the controller that the pods must be annotated

Reply:
I've rechecked the code and seems the pods are not being annotated with this info :thinking:

Reply:
is the namespace who receives the anotations https://github.com/ovn-org/ovn-kubernetes/blob/3606ee31259b93c6cec5e42596b7bec67e2c3bc4/go-controller/pkg/ovn/pods.go#L246-L246

seems ok to remove this checks, sounds like they should not be there since the beginning

Reply:
I think someone who understands the feature being tested needs to sit down with these tests and fix them. It seems likely at this point that these tests are just completely broken and are not testing anything at all, in which case removing the dead code from them is not the right approach, since that code isn't supposed to be dead.

So I think you should just revert any refactoring of these tests. Make whatever changes are needed to get them to compile with 1.19, but don't change anything else.

Reply:
absolutely agree with you